src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -212,49 +258,46 @@ asset::SAssetBundle CImageLoaderTGA::loadAsset(io::IReadFile* _file, const asset
 	region.bufferOffset = 0u;
 	region.bufferImageHeight = 0u;
 	region.imageOffset = { 0u, 0u, 0u };
-	region.imageExtent = image->getCreationParameters().extent;
+	region.imageExtent = imgInfo.extent;
 
 	// read image
 	size_t endBufferSize = {};
 	
 	switch (header.ImageType)
 	{
-		case 1: // Uncompressed color-mapped image
-		case 2: // Uncompressed RGB image
-		case 3: // Uncompressed grayscale image
-			{
-				region.bufferRowLength = calcPitchInBlocks(region.imageExtent.width, getTexelOrBlockBytesize(EF_R8G8B8_SRGB));
-				const int32_t imageSize = endBufferSize = region.imageExtent.height * region.bufferRowLength * bytesPerTexel;
-				texelBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(imageSize);
-				_file->read(texelBuffer->getPointer(), imageSize);
-			}
-			break;
-		
-		case 10: // Run-length encoded (RLE) true color image
+		case STIT_NONE:
+		{
+			os::Printer::log("The given TGA doesn't have image data", _file->getFileName().c_str(), ELL_ERROR);
+			return {};
+		}
+		case STIT_UNCOMPRESSED_COLOR_MAPPED_IMAGE: _IRR_FALLTHROUGH;
+		case STIT_UNCOMPRESSED_RGB_IMAGE: _IRR_FALLTHROUGH;
+		case STIT_UNCOMPRESSED_GRAYSCALE_IMAGE: _IRR_FALLTHROUGH;
+		{
+			region.bufferRowLength = calcPitchInBlocks(region.imageExtent.width, getTexelOrBlockBytesize(EF_R8G8B8_SRGB));
+			const int32_t imageSize = endBufferSize = region.imageExtent.height * region.bufferRowLength * bytesPerTexel;
+			texelBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(imageSize);
+			_file->read(texelBuffer->getPointer(), imageSize);
+		}
+		break;
+		case STIT_RLE_TRUE_COLOR_IMAGE: 
 		{
 			region.bufferRowLength = calcPitchInBlocks(region.imageExtent.width, getTexelOrBlockBytesize(EF_A1R5G5B5_UNORM_PACK16));
```

Comment:
you got the formats for the `getTexelOrBlockBytesize` completely wrong here

Reply:
why? it calculates texels with pitch for `EF_A1R5G5B5_UNORM_PACK16` entry

Reply:
itsn't TRUE COLOR 32bit rgba8 ?

Reply:
actually it depends, but will just take `bytesPerTexel`

Reply:
it may be rgb8 and rgba8

Reply:
it'f fixed